backends/nxp/neutron_partitioner.py

```diff
@@ -0,0 +1,327 @@
+
+    def tag_qdq_clusters(self, nodes: List[torch.fx.Node]):
+        """
+        Identifies QDQ clusters and tag them based on compute operation inside.
+        """
+
+        for node in nodes:
+            if (
+                node.op == "call_function"
+                and not self.is_quant_node(node)
+                and not self.is_dequant_node(node)
+            ):
+                cluster = self.get_qdq_cluster(node)
+                if cluster:
+                    cluster_name = f"{node.name}_cluster"
+                    self.tag_nodes(cluster, cluster_name)
+                    self.cluster_map[cluster_name] = self.QDQCluster(node, cluster)
+
+
+supported_ops = {
+    exir_ops.edge.aten.addmm.default: AddMMConverter,
+    exir_ops.edge.aten.avg_pool2d.default: AvgPool2dConverter,
+    exir_ops.edge.aten.constant_pad_nd.default: ConstantPadNDConverter,
+    exir_ops.edge.aten.convolution.default: ConvolutionConverter,
+    exir_ops.edge.aten.max_pool2d.default: MaxPool2dConverter,
+    exir_ops.edge.aten.max_pool2d_with_indices.default: MaxPool2dConverter,
+    exir_ops.edge.aten.mm.default: MMConverter,
+    exir_ops.edge.aten.relu.default: ReLUConverter,
+    exir_ops.edge.aten._softmax.default: SoftmaxConverter,
+    exir_ops.edge.aten.view_copy.default: ViewCopyConverter,
+}
+
+
+class NeutronSupportedOperators(OperatorSupportBase):
+
+    def __init__(
+        self,
+        qdq_clusters: Dict[str, QDQClusterRecognizer.QDQCluster],
+        target: Target,
+        operators_not_to_delegate: List[str],
+        parameters_mapping: dict[str, Parameter],
+    ):
+        self.qdq_clusters = qdq_clusters
+        self.target = target
+        self.operators_not_to_delegate = operators_not_to_delegate
+        self.parameters_mapping = parameters_mapping
+
+    def _is_node_quantized(self, node: torch.fx.node.Node):
+        return "cluster" in node.meta
+
+    def _is_node_call_function(self, node: torch.fx.node.Node):
+        return node.op == "call_function"
+
+    def is_node_delegatable(self, node: torch.fx.node.Node):
+        if self.operators_not_to_delegate != [""]:
+            any_non_delegatable = any(
+                x in node.name for x in self.operators_not_to_delegate
+            )
+            return not any_non_delegatable
+        return True
+
+    def _is_node_supported_compute(self, node: torch.fx.node.Node) -> bool:
+        """
+        Operator checking function for compute nodes.
+        """
+        if not self.is_node_delegatable(node):
+            return False
+
+        if (node_converter := supported_ops.get(node.target, None)) is None:
+            # There is no `NodeConverter` for this `node`.
+            return False
+
+        return (
+            self._is_node_call_function(node)
+            and self._is_node_quantized(node)
+            and
+            # TODO: `view_copy` node should be delegated only if it's not the only operator in the cluster.
```

Comment:
Is this for Linear to addmm + view_copy etc. decompo? If yes, you may want to use `to_edge_transform_and_lower` with `do-not-decompose` list to not deal with recomposing linear